moveit_ros/moveit_servo/include/moveit_servo/servo_calcs.h

```diff
@@ -274,12 +271,12 @@ class ServoCalcs
 
   moveit::core::RobotStatePtr current_state_;
 
-  // (mutex protected below)
-  // internal_joint_state_ is used in servo calculations. It shouldn't be relied on to be accurate.
-  // original_joint_state_ is the same as incoming_joint_state_ except it only contains the joints the servo node acts
-  // on.
-  // last_joint_state_ caches the previous `original_joint_state_`
-  sensor_msgs::msg::JointState internal_joint_state_, original_joint_state_, last_joint_state_;
+  // These variables are mutex protected
+  // previous_joint_state holds the state q(t - 1)
+  // current_joint_state holds the  state q(t)
+  // next_joint_state holds the computed state q(t + 1)
```

Comment:
We're only iterating forward by one `dt` in each cycle so this comment isn't quite right.

```suggestion
  // current_joint_state holds the  current state as retrieved from the planning scene monitor
  // next_joint_state holds the computed state q(t)
```

Reply:
Oh yea, the `current_joint_state` makes more sense this way.
But `next_joint_state` is still  going to contain the expected future state  `q(t+dt)` right?

Reply:
That's fair. OK, just change `current_joint_state` and I'll be happy.